yarl/_url.py

```diff
@@ -701,11 +683,45 @@ def _validate_authority_uri_abs_path(host, path):
                 "Path in a URL with authority should start with a slash ('/') if set"
             )
 
+    def _make_child(self, segments, encoded=False):
+        # add segments to self._val.path, accounting for absolute vs relative paths
+        parsed = []
+        for seg in reversed(segments):
+            if not seg:
+                continue
+            if seg[0] == "/":
+                raise ValueError(
+                    f"Appending path {seg!r} starting from slash is forbidden"
+                )
+            seg = seg if encoded else self._PATH_QUOTER(seg)
+            if "/" in seg:
+                parsed += (
+                    sub for sub in reversed(seg.split("/")) if sub and sub != "."
+                )
+            elif seg != ".":
+                parsed.append(seg)
+        parsed.reverse()
+        old_path = self._val.path
+        if old_path:
+            parsed = [*old_path.rstrip("/").split("/"), *parsed]
+        if self.is_absolute():
+            parsed = self._normalize_path_segments(parsed)
+        new_path = "/".join(parsed)
+        return URL(
+            self._val._replace(path=new_path, query="", fragment=""), encoded=True
+        )
+
     @classmethod
     def _normalize_path(cls, path):
-        # Drop '.' and '..' from path
+        # Drop '.' and '..' from str path
 
         segments = path.split("/")
+        return "/".join(cls._normalize_path_segments(segments))
+
+    @classmethod
+    def _normalize_path_segments(cls, segments):
```

Comment:
This shouldn't be a `@classmethod`. It better be a pure function, or if it really needs to be under the class, use `@staticmethod` since `cls` is unused.

Reply:
I was following the general style of the class here; e.g. `_encode_host()` can also be made a staticmethod.